File: TasksList/Node.js/Web/Plain/src/storage/DatabaseFactory.js
const PostgreSQLAdapter = require('./adapters/PostgreSQL');
const MySQLAdapter = require('./adapters/MySQL');
const SQLiteAdapter = require('./adapters/SQLite');
const SQLServerAdapter = require('./adapters/SQLServer');
const MongoDBAdapter = require('./adapters/MongoDB');

function createAdapter(defaultDB) {
  const driver = process.env.DB_DRIVER || 'pgsql';
  switch (driver) {
    case 'sqlite':
      return new SQLiteAdapter(defaultDB);
    case 'mysql':
      return new MySQLAdapter(defaultDB);
    case 'mongodb':
      return new MongoDBAdapter(defaultDB);
    case 'sqlserver':
    case 'mssql':
      return new SQLServerAdapter(defaultDB);
    case 'pgsql':
    default:
      return new PostgreSQLAdapter(defaultDB);
  }
}

module.exports = { createAdapter };

File: Inboxes/Python/Web/Plain/src/storage/database_factory.py
import os
from storage.adapters.postgresql import PostgreSQL
from storage.adapters.mysql import MySQL
from storage.adapters.sqlite import SQLite
from storage.adapters.sqlserver import SQLServer
from storage.adapters.mongodb import MongoDB

def create_adapter():
    driver = os.getenv("DB_DRIVER", "pgsql")
    if driver == "sqlite":
        return SQLite()
    if driver in ("sqlserver", "mssql"):
        return SQLServer()
    if driver == "mongodb":
        return MongoDB()
    if driver == "mysql":
        return MySQL()
    return PostgreSQL()
